### ISSUE TITLE
Add a RNN Predictor

### ISSUE THREAD

**Issue by NIlS-Z99** [NIlS-Z99/Invest-Tool#4]
something akin to:
`import yfinance as yf
import numpy as np
import pandas as pd
import tensorflow as tf
from sklearn.preprocessing import MinMaxScaler

# Collect historical stock data
ticker = "AAPL"
data = yf.download(ticker, start="2016-01-01", end="2021-12-31")

# Preprocess the data
data.dropna(inplace=True)
scaler = MinMaxScaler()
data['Close'] = scaler.fit_transform(data[['Close']])
data['Volume'] = scaler.fit_transform(data[['Volume']])

# Split the data into train and test sets
train_size = int(len(data) * 0.8)
train_data, test_data = data.iloc[:train_size], data.iloc[train_size:]

# Create time series data
def create_time_series(data, window_size):
    X, y = [], []
    for i in range(len(data)-window_size):
        X.append(data[i:i+window_size])
        y.append(data[i+window_size])
    return np.array(X), np.array(y)

window_size = 20
X_train, y_train = create_time_series(train_data['Close'].values.reshape(-1, 1), window_size)
X_test, y_test = create_time_series(test_data['Close'].values.reshape(-1, 1), window_size)

# Create RNN model
model = tf.keras.Sequential([
    tf.keras.layers.LSTM(128, input_shape=(window_size, 1)),
    tf.keras.layers.Dense(64, activation='relu'),
    tf.keras.layers.Dense(1)
])

model.compile(optimizer='adam', loss='mse')

# Train model
model.fit(X_train, y_train, epochs=100, validation_data=(X_test, y_test))

# Make predictions for the next year
last_date = data.index[-1]
next_year = pd.date_range(start=last_date, periods=365, freq='D')[1:]
last_20_days = data['Close'].iloc[-20:].values.reshape(-1, 1)

next_year_predictions = []
for i in range(len(next_year)):
    input_data = scaler.transform(last_20_days)
    input_data = input_data.reshape((1, window_size, 1))
    prediction = model.predict(input_data)
    next_year_predictions.append(scaler.inverse_transform(prediction)[0])
    last_20_days = np.concatenate((last_20_days[1:], prediction), axis=0)

# Output predictions
next_year_predictions = pd.DataFrame(next_year_predictions, columns=['Close'], index=next_year)
print("Predictions for the next year:")
print(next_year_predictions)`
